src/pages/ReimbursementAccount/BeneficialOwnerInfo/BeneficialOwnerInfo.js

```diff
@@ -0,0 +1,301 @@
+                      city: lodashGet(reimbursementAccountDraft, `beneficialOwner_${ownerKey}_city`),
+                      state: lodashGet(reimbursementAccountDraft, `beneficialOwner_${ownerKey}_state`),
+                      zipCode: lodashGet(reimbursementAccountDraft, `beneficialOwner_${ownerKey}_zipCode`),
+                  }));
+
+        BankAccounts.updateBeneficialOwnersForBankAccountDraft({
+            ownsMoreThan25Percent: isUserUBO,
+            beneficialOwners: JSON.stringify(beneficialOwners),
+            beneficialOwnerKeys,
+            bankAccountID,
+        });
+        setIsBeneficialOwnerInfoSet(true);
+    };
+
+    const addBeneficialOwner = (beneficialOwnerID) => {
+        // Each beneficial owner is assigned a unique key that will connect it to values in saved ONYX.
+        // That way we can dynamically render each Identity Form based on which keys are present in the beneficial owners array.
+        setBeneficialOwnerKeys((previousBeneficialOwners) => {
+            const newBeneficialOwners = [...previousBeneficialOwners, beneficialOwnerID];
+            FormActions.setDraftValues(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {beneficialOwners: JSON.stringify(newBeneficialOwners)});
+            return newBeneficialOwners;
+        });
+    };
+
+    const startFrom = 0;
+    const handleBeneficialOwnerDetailsFormSubmit = () => {
+        if (_.find(beneficialOwnerKeys, (beneficialOwnerID) => beneficialOwnerID === beneficialOwnerBeingModifiedID) === undefined && canAddMoreUBOS) {
+            addBeneficialOwner(beneficialOwnerBeingModifiedID);
+        }
+
+        // Because beneficialOwnerKeys array is not yet updated at this point we need to check against lower MAX_NUMBER_OF_UBOS (account for the one that is being added)
+        const isLastUBOThatCanBeAdded = beneficialOwnerKeys.length === (isUserUBO ? MAX_NUMBER_OF_UBOS - 2 : MAX_NUMBER_OF_UBOS - 1);
+        setCurrentUBOSubstep(isEditingCreatedBeneficialOwner || isLastUBOThatCanBeAdded ? substep.UBOS_LIST : substep.ARE_THERE_MORE_UBOS);
+        setIsEditingCreatedBeneficialOwner(false);
+    };
+
+    const {
+        componentToRender: BeneficialOwnerDetailsForm,
+        isEditing,
+        screenIndex,
+        nextScreen,
+        prevScreen,
+        moveTo,
+        resetScreenIndex,
+    } = useSubStep({bodyContent, startFrom, onFinished: handleBeneficialOwnerDetailsFormSubmit});
+
+    const prepareBeneficialOwnerDetailsForm = () => {
+        const beneficialOwnerID = Str.guid();
+        setBeneficialOwnerBeingModifiedID(beneficialOwnerID);
+        // Reset Beneficial Owner Details Form to first substep
+        resetScreenIndex();
+        setCurrentUBOSubstep(substep.UBO_DETAILS_FORM);
+    };
+
+    const handleNextUBOSubstep = (value) => {
+        if (currentUBOSubstep === substep.IS_USER_UBO) {
+            setIsUserUBO(value);
+
+            // user is an owner but there are 4 other owners already added, so we remove last one
```

Comment:
Same for other comments.
```suggestion
            // User is an owner but there are 4 other owners already added, so we remove last one
```